tests/integration/vxlan_test.py

```diff
@@ -0,0 +1,179 @@
+
+
+def test_add_and_remove_vxlan(eth1_up):
+    with vxlan_interface(
+        VXLAN1_IFNAME, VXLAN1_ID, eth1_up[Interface.KEY][0][Interface.NAME]
+    ) as desired_state:
+        assertlib.assert_state(desired_state)
+
+    current_state = statelib.show_only((VXLAN1_IFNAME,))
+    assert not current_state[Interface.KEY]
+
+
+@pytest.fixture
+def vxlan_on_eth1(eth1_up):
+    with vxlan_interface(
+        VXLAN1_IFNAME, VXLAN1_ID, eth1_up[Interface.KEY][0][Interface.NAME]
+    ) as desired_state:
+        parent_name = desired_state[Interface.KEY][0][VXLAN.CONFIG_SUBTREE][
+            VXLAN.BASE_IFACE
+        ]
+        iface_states = statelib.show_only((parent_name, VXLAN1_IFNAME))
+        yield iface_states
+
+
+def test_add_and_remove_two_vxlans_on_same_iface(eth1_up):
+    with two_vxlans_on_eth1() as desired_state:
+        assertlib.assert_state(desired_state)
+
+    vxlan_interfaces = [i['name'] for i in desired_state[Interface.KEY]]
+    current_state = statelib.show_only(vxlan_interfaces)
+    assert not current_state[Interface.KEY]
+
+
+def test_rollback_for_vxlans(eth1_up):
+    current_state = libnmstate.show()
+    desired_state = TWO_VXLANS_STATE
+
+    desired_state[Interface.KEY][1]['invalid_key'] = 'foo'
+    with pytest.raises(NmstateVerificationError):
+        libnmstate.apply(desired_state)
+
+    time.sleep(5)  # Give some time for NetworkManager to rollback
+    current_state_after_apply = libnmstate.show()
+    assert current_state == current_state_after_apply
+
+
+def test_set_vxlan_iface_down(eth1_up):
+    with vxlan_interface(
+        VXLAN1_IFNAME, VXLAN1_ID, eth1_up[Interface.KEY][0][Interface.NAME]
+    ):
+        libnmstate.apply(
+            {
+                Interface.KEY: [
+                    {'name': VXLAN1_IFNAME, 'type': 'vxlan', 'state': 'down'}
+                ]
+            }
+        )
+
+        current_state = statelib.show_only((VXLAN1_IFNAME,))
+        assert not current_state[Interface.KEY]
+
+
+def test_add_new_base_iface_with_vxlan():
+    parent = 'dummy00'
+    desired_state = {
+        Interface.KEY: [
+            {
+                Interface.NAME: '{}.{}'.format(parent, VXLAN1_ID),
+                Interface.TYPE: InterfaceType.VXLAN,
+                Interface.STATE: InterfaceState.UP,
+                VXLAN.CONFIG_SUBTREE: {
+                    VXLAN.ID: VXLAN1_ID,
+                    VXLAN.BASE_IFACE: parent,
+                    VXLAN.REMOTE: '192.168.100.2',
+                },
+            },
+            {
+                Interface.NAME: parent,
+                Interface.TYPE: InterfaceType.DUMMY,
+                Interface.STATE: InterfaceState.UP,
+            },
+        ]
+    }
+    try:
+        libnmstate.apply(desired_state)
+    finally:
+        for ifstate in desired_state[Interface.KEY]:
+            ifstate[Interface.STATE] = InterfaceState.ABSENT
+        libnmstate.apply(desired_state, verify_change=False)
+
+
+@contextmanager
+def two_vxlans_on_eth1():
```

Comment:
This smells duplication with the `vxlan_interface` helper.
You can create to states without `apply` and then simply merge them. OR you can adjust the helper to create multiple vxlan ifaces on the same base. The first sounds nicer to me, especially when the later is used only once.